src/parser.rs

```diff
@@ -537,3 +585,38 @@ fn test_environment_validation() {
     );
     assert_eq!(validate_environment("good"), Ok(()));
 }
+
+#[test]
+fn test_version_ordering() {
+    macro_rules! ver {
+        ($v:expr) => {
+            Version::parse($v).unwrap()
+        };
+    }
+
+    assert_eq!(ver!("1.0.0"), ver!("1.0.0"));
+    assert_ne!(ver!("1.1.0"), ver!("1.0.0"));
+    assert_eq!(ver!("1.0"), ver!("1.0.0"));
+    assert_eq!(ver!("1.0dev"), ver!("1.0.0-dev"));
+    assert_eq!(ver!("1.0dev"), ver!("1.0.0.0-dev"));
+    assert_ne!(ver!("1.0dev"), ver!("1.0.0.0-dev1"));
+
+    assert!(ver!("1.0dev") >= ver!("1.0.0.0-dev"));
+    assert!(ver!("1.0dev") < ver!("1.0.0.0"));
+
+    assert!(ver!("1.0.0") > ver!("1.0.0-rc1"));
+    assert!(ver!("1.0.0") >= ver!("1.0.0-rc1"));
```

Comment:
I don't understand why line 608 would pass, it should fail no?

Reply:
why should it fail? `1.0.0` is greater or equal to the pre-release version `1.0.0-rc1`.

Reply:
Shouldn't it be just greater than?

Reply:
it's both :)